### PR TITLE
DrtRoute should provide the unshared ride path

### PR DESCRIPTION
**Pull request by steffenaxer**
This PR is a proposal for #1566

### CODE REVIEW

**Comment by sebhoerl** [comment]
Just as a comment: This now replicates the same set-up as we have in `DefaultPassengerTransitRoute` for public transport, which is nice as things become more streamlined and extensible.

One remark: Maybe we should have a fallback for old-format route descriptions, e.g. either
- Testing the description with a regex like `[0-9]+ [0-9]+`
- Testing whether description starts with `{`

And creating a deprecation warning if old-format route is used (and then to be removed in later version). But let's first see what everybody else says :)

**Comment by steffenaxer** [comment]
> Just as a comment: This now replicates the same set-up as we have in `DefaultPassengerTransitRoute` for public transport, which is nice as things become more streamlined and extensible.
> 
> One remark: Maybe we should have a fallback for old-format route descriptions, e.g. either
> 
>     * Testing the description with a regex like `[0-9.]+ [0-9.]+`
> 
>     * Testing whether description starts with `{`
> 
> 
> And creating a deprecation warning if old-format route is used (and then to be removed in later version). But let's first see what everybody else says :)

@sebhoerl, I can't find code that reads directly from the RouteDescription. Do you have any code, related to DMC that parses the RouteDescription? Maybe you can point me to a part, where the information from the RouteDescription is parsed and used. In this case, I will try to implement a retrofit version.

File: contribs/drt/src/main/java/org/matsim/contrib/drt/routing/DrtRoute.java [hunk 1]
```diff
@@ -39,44 +47,60 @@
...
-		return maxWaitTime.seconds() + " " + directRideTime.seconds();
+		try {
+			return new ObjectMapper().writeValueAsString(routeDescription);
+		} catch (JsonProcessingException e) {
+			throw new RuntimeException(e);
+		}
 	}
 
 	@Override
 	public void setRouteDescription(String routeDescription) {
```
Comment (sebhoerl) [inline contribs/drt/src/main/java/org/matsim/contrib/drt/routing/DrtRoute.java:98]: @steffenaxer This is called when the route is read from a plans file. So here could be a condition to fall back to the old format in `routeDescription`.

Comment (steffenaxer) [reply contribs/drt/src/main/java/org/matsim/contrib/drt/routing/DrtRoute.java:98]: Ok, got it. I will provide a retrofit version

**Review by michalmac — APPROVED** [review]
⭐ 

Re original request in #1566: I do not see any reason for not providing the unshared path :-)

**Comment by steffenaxer** [comment]
I need to test if things are working now when reading an old plans file. There is currently no test that reads in plans that already contains routed drt plans.

**Comment by sebhoerl** [comment]
I think it will be enough to have a test for both formats on `setRouteDescription`

**Comment by steffenaxer** [comment]
> I think it will be enough to have a test for both formats on `setRouteDescription`, even better without a static plans file (whose format may change in the future).

I added a commit that contains a handling for old and new format. @sebhoerl maybe you could have a look?

**Review by sebhoerl — APPROVED** [review]

**Comment by sebhoerl** [comment]
Nice, but I think the test is not committed yet, is it?

**Comment by steffenaxer** [comment]
Commit dee1acf92b56459a4133bb6b137e98fb2848bdc1 contains your proposed implementation. I haven't implemented a warning. Do you think we need something?

**Comment by sebhoerl** [comment]
Ah, sorry. I meant to add a unit test for `DrtRoute::setRouteDescription`.

**Comment by steffenaxer** [comment]
Ok, I assumed it, but was not 100% sure. I will add a unit test tomorrow. Thanks for your help!

**Comment by steffenaxer** [comment]
@sebhoerl junit test is implemented. Please have a look.

**Pull request merged**